Use the `options` parameter of `Date.prototype.toLocaleDateString` method to set date content

nobuoka opened the pull request:
## Problem

There is the problem that a wrong date string can be shown in the previous way.

E.g. `<time>1912-01-01</time>` shows `1911-12-31` in the time zone Asia/Seoul.

## Cause

The reason is that there are the time zones in which past offsets and current offsets are different.

For example, the time offset of Korea was changed from UTC+08:30 to UTC+09:00 in 1912. So that in that time zone,

* `new Date(1912, 0, 1).toLocaleDateString()` results in `"1911/12/31"` while
* `new Date(1913, 0, 1).toLocaleDateString()` results in `"1913/01/01"`.

## The way to fix

To fix this problem, a second parameter of `Date.prototype.toLocaleDateString` method is used.

* [ECMA-402 (1st)](http://www.ecma-international.org/ecma-402/1.0/#sec-13.3.2)

This way was proposed by @wakaba. 

### Check test on several user agents

#### PC

To check that way can be used or not on each user agent, I used my PC (Windows 10) and [browserling](https://www.browserling.com/internet-explorer-testing), which provides several user agents through the web browser. Test code is [https://jsfiddle.net/wdem535u/4/](https://jsfiddle.net/wdem535u/4/).

* Windows 10, Edge 40.15063.0.0 : OK
* Windows 10, IE 11.0.42 : OK
* Windows 10, Firefox 53.0.3 : OK
* Windows 7, Safari 5.1 : **NG (it shows `Monday, December 30, 999`)**
* Windows 7, Chrome 58 : OK (it shows `1/1/1000`)
* Windows Vista, IE 9 : OK (it shows `1/1/1000`)
* Windows Vista, Opera 36 : OK (it shows `1/1/1000`)

Safari 5.1 is too old, so I think it's not problem that Safari 5.1 can't handle this way.

#### Mobile

According to [MDN's `Date#toLocaleDateString` page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString), most mobile browsers doesn't support locales and options parameters of `Date.prototype.toLocaleDateString`. (Only Chrome for Android 26+ and Safari Mobile 10+ do.)

I think this is problem.

nobuoka committed:
Use `Date.prototype.toLocaleDateString` method to set date content

There is the problem that wrong date can be shown in the previous way.
(There are the time zones in which past offsets and current offsets
are different.)

wakaba commented on src/time.js:
```diff
@@ -325,7 +318,7 @@ is a willful violation to the current HTML Living Standard.
 /* ***** BEGIN LICENSE BLOCK *****
- * Copyright 2008-2015 Wakaba <wakaba@suikawiki.org>.  All rights reserved.
+ * Copyright 2008-2017 Wakaba <wakaba@suikawiki.org>.  All rights reserved.
  *
```
Insert ```Copyright 2017 Hatena <http://hatenacorp.jp/>``` here.

nobuoka committed:
The time value of `Date` object representing date is set to 12:00 UTC

This is for old browsers, which don't support the options parameter
of `toLocaleDateString` method.

nobuoka committed:
Copyright
